dhis-2/dhis-api/src/main/java/org/hisp/dhis/programrule/engine/ProgramRuleEngineService.java

```diff
@@ -37,7 +39,11 @@
  */
 public interface ProgramRuleEngineService
 {
-    List<RuleEffect> evaluateEnrollment( long enrollment );
+    List<RuleEffect> evaluateEnrollment( ProgramInstance enrollment );
```

Comment:
The interface now uses objects and `long`, which is a bit confusing. I normally favour primary keys as function parameters: when using Hibernate there is no overhead anyway, because the object is normally in the first level cache, so the additional lookup is free.

Reply:
Changed to use long and String.
We need to use the id in the listener, because is how the event is created and from the importer I do not have the id (at least in a simple way), so we would use the UID